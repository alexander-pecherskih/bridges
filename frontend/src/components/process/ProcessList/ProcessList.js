import React from 'react'
import PropTypes from 'prop-types'

import TBodyMessage from '../../common/TBodyMessage'
import { Link } from 'react-router-dom'

const ProcessRow = ({ process }) => {
  return <tr>
    <td>
      { process.id }
            &nbsp;-&nbsp;
      <Link to={`/process/${process.id}`}>diagram</Link>
    </td>
    <td>{ process.title }</td>
  </tr>
}

ProcessRow.propTypes = {
  process: PropTypes.object
}

const ProcessList = ({ processes, loading, errorMessage }) => {
  let rows = processes.map((item) => {
    return <ProcessRow process={ item } key={ item.id }/>
  })

  if (loading) {
    rows = <TBodyMessage colSpan={ 2 } message="Loading..." />
  } else if (rows.length === 0) {
    rows = <TBodyMessage colSpan={ 2 } message="No one Process Found" />
  }

  if (errorMessage) {
    rows = <TBodyMessage colSpan={ 2 } message={ errorMessage } />
  }

  return <table>
    <thead>
      <tr>
        <th>id</th>
        <th>title</th>
      </tr>
    </thead>
    <tbody>
      { rows }
    </tbody>
  </table>
}

ProcessList.propTypes = {
  processes: PropTypes.array.isRequired,
  loading: PropTypes.bool.isRequired,
  errorMessage: PropTypes.string
}

export default ProcessList
