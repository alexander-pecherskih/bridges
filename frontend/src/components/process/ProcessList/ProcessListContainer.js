import React, { useEffect } from 'react'
import PropTypes from 'prop-types'
import { compose } from 'redux'
import { connect } from 'react-redux'

import { getProcesses } from '../../../store/actions/processList'
import ProcessList from './ProcessList'

const ProcessListContainer = ({ processes, loading, error, getProcesses }) => {
  const onMount = () => {
    getProcesses()
  }
  useEffect(onMount, [])

  return (
    <ProcessList processes={processes} loading={loading} errorMessage={error} />
  )
}

ProcessListContainer.propTypes = {
  processes: PropTypes.array.isRequired,
  loading: PropTypes.bool.isRequired,
  error: PropTypes.object,
  getProcesses: PropTypes.func.isRequired,
}

const mapStateToProps = ({ processList: { processes, loading, error } }) => {
  return { processes, loading, error }
}

const mapDispatchToProps = (dispatch) => {
  return {
    getProcesses: () => dispatch(getProcesses()),
  }
}

export default compose(connect(mapStateToProps, mapDispatchToProps))(
  ProcessListContainer
)
