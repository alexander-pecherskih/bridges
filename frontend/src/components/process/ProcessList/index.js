import ProcessListContainer from './ProcessListContainer'

export default ProcessListContainer