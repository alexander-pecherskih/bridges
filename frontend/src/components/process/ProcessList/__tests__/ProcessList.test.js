import React from 'react'
import { shallow } from 'enzyme'

import ProcessList from '../ProcessList'

const processes = [
  {
    id: '1',
    title: 'process 1',
  },
  {
    id: '2',
    title: 'process 2',
  },
]

describe('<ProcessList />', () => {
  it('ProcessList initial', () => {
    const wrapper = shallow(
      <ProcessList processes={processes} loading={false} errorMessage={null}/>
    )

    expect(wrapper).toMatchSnapshot();
  })

  it('ProcessList loading', () => {
    const wrapper = shallow(
      <ProcessList processes={processes} loading={true} />
    )

    expect(wrapper).toMatchSnapshot();
  })

  it('ProcessList is empty', () => {
    const wrapper = shallow(<ProcessList processes={[]} loading={false} />)

    expect(wrapper).toMatchSnapshot();
  })
})
