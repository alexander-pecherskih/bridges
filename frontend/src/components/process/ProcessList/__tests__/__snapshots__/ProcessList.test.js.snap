// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProcessList /> ProcessList initial 1`] = `ShallowWrapper {}`;

exports[`<ProcessList /> ProcessList is empty 1`] = `ShallowWrapper {}`;

exports[`<ProcessList /> ProcessList loading 1`] = `ShallowWrapper {}`;
